### ISSUE TITLE
Standardize define api

### ISSUE THREAD

**Issue by stambata** [softwaregroup-bg/ut-error#5]
Let's decide how exactly we want uterror.define to work.

**Comment by stambata** [comment]
utError.define accepts 4 arguments:

name - the name of the error
superType - error class' super constructor (UTError by default). (used for building the error.type by recursively concatenating error names until base class is reached)
defaultMessage - error's default message (currently it can not be overridden once set)
errorObject - I don't even know how this appeared in ut-error and what it is used for. I think somebody misunderstood how ut-error works and used that to inject errors directly.

Anyaway currently we have the following 4 cases for an argument that we can pass to the CustomUTError constructor:

* a real javascript exception.
  * It gets set as a cause.
  * The error message will be overriden by the defaultMessage (in case it is defined). Do we want that?
  * Interpolation can't be applied as there is no params object. 
* a string
  * the string just gets set as a message.
  * Interpolation can't be applied as there is no params object. 
* an object
  * if obj.message is passed it will be set as an error message (unless a defaultMessage is set which will override it).
  * Interpolation can be applied if there's params object. 
* anything else
  * a generic error will be thrown with message: 'Unknown error.'

**Can we summarize how we want it to work?**

Basically there are several main points to consider:

1) How do we want to treat the defaultMessage?
  1.1) Always override the error message (the way it is now)?
  1.2) Be used as a fallback only in case there's no message
2) Do we want the default message to be mandatory?
  2.1) If we chose 1.1) then maybe we do
  2.2) if we chose 1.2) then maybe we don't
3) Do we want to be able to pass a string as an argument to the CustomUTError constructor?
  3.1) If we chose 2.1) then maybe we don't
  3.2) if we chose 2.2) then maybe we do
  

@kalinkrustev @zetxx

**Comment by kalinkrustev** [comment]
Decision is:

1) defaultMessage is required (rename to message) in case there is superType passed
2) errorObject is not needed
3) provide way to define level
4) check if name contains '.'
5) when creating error, can only pass object or Error or undefined

**Comment by kalinkrustev** [comment]
resolved by #6
